doc/manual/src/language/advanced-attributes.md

```diff
@@ -2,6 +2,26 @@
 
 Derivations can declare some infrequently used optional attributes.
 
+  - [`referenceablePaths`]{#adv-attr-referenceablePaths}\
+    The optional attribute `referenceablePaths` restricts the set of
+    store paths whose cryptographic hash Nix looks for in the build
+    outputs in order to establish runtime dependencies. By default,
+    this set includes build inputs, output paths, and paths built by
+    recursive Nix invocations.
+    If this attribute is specified, then this set is restricted (in the
+    sense of set intersection) to the given list of paths.
+    For example,
```

Comment:
I find this doc a bit too low-level. Maybe something around the lines of

```suggestion
  - [`referenceablePaths`]{#adv-attr-referenceablePaths}\
    The optional attribute `referenceablePaths` restricts the set of
    store paths that Nix will consider when scanning for the runtime
    dependencies.
    If the attribute is unset, then this set will consist of all the build
    inputs, output paths, and paths built by recursive Nix invocations
    
    For example,
```

(But it's great to have some doc without even having to ask for it! 😃 )